src/LinuxTracing/TracerImpl.cpp

```diff
@@ -294,35 +322,61 @@ bool TracerImpl::OpenUserSpaceProbes(const std::vector<int32_t>& cpus) {
     AddUretprobesFileDescriptors(uretprobes_fds_per_cpu, function);
     AddUprobesFileDescriptors(uprobes_fds_per_cpu, function);
 
-    for (const auto& [cpu, fd] : uretprobes_fds_per_cpu) {
-      uprobes_uretpobres_fds_per_cpu[cpu].push_back(fd);
-    }
-    for (const auto& [cpu, fd] : uprobes_fds_per_cpu) {
-      uprobes_uretpobres_fds_per_cpu[cpu].push_back(fd);
-    }
+    OpenRingBuffersOrRedirectOnExisting(uretprobes_fds_per_cpu, &fds_per_cpu_for_redirection,
+                                        &ring_buffers_, UPROBES_RING_BUFFER_SIZE_KB,
+                                        "uprobes_uretprobes");
+    OpenRingBuffersOrRedirectOnExisting(uprobes_fds_per_cpu, &fds_per_cpu_for_redirection,
+                                        &ring_buffers_, UPROBES_RING_BUFFER_SIZE_KB,
+                                        "uprobes_uretprobes");
```

Comment:
I hope the behavior is the same as before. It looks ok but it's hard to tell with confidence. If you haven't yet, please take some captures with multiple uprobes and check that all functions get instrumented correctly.